app/scripts/metamask-controller.js

```diff
@@ -1628,17 +1628,14 @@ export default class MetamaskController extends EventEmitter {
       onboardingController: this.onboardingController,
       controllerMessenger: this.controllerMessenger.getRestricted({
         name: 'AccountTracker',
+        allowedActions: ['AccountsController:getSelectedAccount'],
```

Comment:
```suggestion
        allowedActions: ['AccountsController:getSelectedAccount', 'NetworkController:getNetworkClientById'],
```
Shouldn't we need `'NetworkController:getNetworkClientById'` in the `allowedActions` ?

Reply:
I agree that this should have been included earlier to my changes. However, the action call is not currently used in the present state of the controller usage in extension. I did, however, add the action in this [PR](https://github.com/MetaMask/metamask-extension/pull/27258/files), which converts the same controller to BaseControllerV2. In this version, we rely solely on actions and events to interact with the NetworkController. The other PR will be available for review once this one is merged.